app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -275,8 +275,9 @@ def get_hardware_network_info(instance)
             public_ip_obj = ipconfig.properties.try(:public_ip_address)
             next unless public_ip_obj
 
-            name = File.basename(public_ip_obj.id)
-            ip_profile = @ips.get(name, nic_profile.resource_group)
+            ip_profile = ip_addresses.find { |ip| ip.id == public_ip_obj.id }
```

Comment:
hm, weird, the spec you wrote should not be connected to this code. Spec is connected to the refresh parser in NetworkManager, so it should work without this change?

Comment of this methods says that hardware/network will go away. I need to do a final check, where is this model being used. But e.g. the ip_addresses are now being loaded from network manager, after this PR https://github.com/ManageIQ/manageiq/pull/8445 , only google uses the old model now.

So probably, all the network related code, could be deleted from cloud_manager/refresh_parser now.